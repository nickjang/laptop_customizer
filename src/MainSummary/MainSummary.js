import React, { Component } from 'react';
import Option from '../Option/Option';
import Total from '../Total/Total';
import './MainSummary.css'

class MainSummary extends Component {
  static defaultProps = {
    selected: []
  };
  
  render() {
    const summary = Object.keys(this.props.selected).map((feature, idx) => {
      const featureHash = feature + '-' + idx;
      return <Option key={featureHash} feature={feature} idx={idx} selectedOption={this.props.selected[feature]} />;
    });

    return (
      <section className="main__summary">
        <h2>Your cart</h2>
        {summary}
        <Total selected={this.props.selected} />
      </section>
    );
  }
}

export default MainSummary;