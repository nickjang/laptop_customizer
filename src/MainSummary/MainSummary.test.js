import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MainSummary from './MainSummary';

const selected = {
  Processor: {
    name: '17th Generation Intel Core HB (7 Core with donut spare)',
    cost: 700
  },
  'Operating System': {
    name: 'Ubuntu Linux 16.04',
    cost: 200
  },
  'Video Card': {
    name: 'Toyota Corolla 1.5v',
    cost: 1150.98
  },
  Display: {
    name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
    cost: 1500
  }
}

describe('<MainSummary />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MainSummary />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders empty when no props supplied', () => {
    const wrapper = shallow(<MainSummary />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders MainSummary with selected options when props are supplied', () => {
    const wrapper = shallow(
      <MainSummary selected={selected} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});