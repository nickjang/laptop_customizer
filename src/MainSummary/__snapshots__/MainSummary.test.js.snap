// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainSummary /> renders MainSummary with selected options when props are supplied 1`] = `
<section
  className="main__summary"
>
  <h2>
    Your cart
  </h2>
  <Option
    feature="Processor"
    idx={0}
    key="Processor-0"
    selectedOption={
      Object {
        "cost": 700,
        "name": "17th Generation Intel Core HB (7 Core with donut spare)",
      }
    }
  />
  <Option
    feature="Operating System"
    idx={1}
    key="Operating System-1"
    selectedOption={
      Object {
        "cost": 200,
        "name": "Ubuntu Linux 16.04",
      }
    }
  />
  <Option
    feature="Video Card"
    idx={2}
    key="Video Card-2"
    selectedOption={
      Object {
        "cost": 1150.98,
        "name": "Toyota Corolla 1.5v",
      }
    }
  />
  <Option
    feature="Display"
    idx={3}
    key="Display-3"
    selectedOption={
      Object {
        "cost": 1500,
        "name": "15.6\\" UHD (3840 x 2160) 60Hz Bright Lights and Knobs",
      }
    }
  />
  <Total
    selected={
      Object {
        "Display": Object {
          "cost": 1500,
          "name": "15.6\\" UHD (3840 x 2160) 60Hz Bright Lights and Knobs",
        },
        "Operating System": Object {
          "cost": 200,
          "name": "Ubuntu Linux 16.04",
        },
        "Processor": Object {
          "cost": 700,
          "name": "17th Generation Intel Core HB (7 Core with donut spare)",
        },
        "Video Card": Object {
          "cost": 1150.98,
          "name": "Toyota Corolla 1.5v",
        },
      }
    }
  />
</section>
`;

exports[`<MainSummary /> renders empty when no props supplied 1`] = `
<section
  className="main__summary"
>
  <h2>
    Your cart
  </h2>
  <Total
    selected={Array []}
  />
</section>
`;
